src/main/java/it/gov/pagopa/payment/service/payment/idpaycode/IdpayCodePreAuthServiceImpl.java

```diff
@@ -76,14 +77,13 @@ public RelateUserResponse relateUser(String trxId, String fiscalCode) {
     public AuthPaymentDTO previewPayment(String trxId, String merchantId) {
 
         TransactionInProgress trx = transactionInProgressRepository.findById(trxId)
-                .orElseThrow(() -> new ClientExceptionWithBody(
-                        HttpStatus.NOT_FOUND,
+                .orElseThrow(() -> new TransactionNotFoundOrExpiredException(
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor